### ISSUE TITLE
AUR package

### ISSUE THREAD

**Issue by ghost** [CodesOfRishi/smartcd#3]
Do you have plans to package it for the AUR?

I can offer 2 options:

- I can help you package it or package it for you and provide a rundown of how to put it on AUR. 
- I can put it on AUR and maintain it myself.

Let me know if you want either or none. 

Cheers

**Comment by CodesOfRishi** [comment]
Never dealt with Arch related stuffs but eager to know more. The 1st offer seems ok to me, as per learning perspective. Many thanks for that.
I would like to package it myself with help, obviously. In such a way I will get to know things as well.

Also, I read the Arch Wiki page [here](https://wiki.archlinux.org/title/PKGBUILD) regarding the `PKGBUILD`. The variables `pkgname`, `pkgver`, `pkgrel`, and `arch` are mandatory, but I have never versioned `SmartCd`, also there have been a lot of commits. So going back to versioning it from the start seems a hectic task.

**Comment by theAkito** [comment]
> I have never versioned `SmartCd`, also there have been a lot of commits. So going back to versioning it from the start seems a hectic task.

Not sure, where you are getting at. From what I know, when packaging software in general, if you never versioned it before, you just version the newest stable release you have available (in your case `HEAD`, I suppose) and release that one. No need to release older versions, except there is a true necessity in releasing an older version, which I do not see here.
Contrary to your fears, it's usually not recommended to release older versions newly packaged, in the first place, if you have a newer version available, which is just better than the old ones.

All that is taken from my experience with different packaging systems/authorities and I have yet to experience how exactly the AUR works, so it might be different in this specific case. However, I strongly doubt you would need to release older versions of this, when it would, as far as I know, make no sense, in the slightest, to do that.

**Comment by CodesOfRishi** [comment]
Yes exactly! I too do not know how exactly the AUR works, that's why I thought to just mention it! 
I have tagged a few commits with versions today. I have locally, will push it later on.

Thanks 👍

**Comment by ghost** [comment]
You don't even need a version. A common practice is to generate a version from git commits like:

```
pkgver() {
  cd "${srcdir}/${pkgname}"
  printf "r%s.%s" "$(git rev-list --count HEAD)" "$(git rev-parse --short HEAD)"
}
```
(This goes inside the PKGBUILD file)

This will automatically set the `pkgver` to something in the form of `r17.abc3def` where `17` is the total number of commits so far and `abc3def` is the short hash of the last commit.

___

If you are not sure about how pushing a package to AUR work (at least for me it was unclear from reading the wiki) I have written a short step by step guide for a different package, which you could take a look at: https://github.com/Pocco81/MiniPrompt/pull/2

**Comment by CodesOfRishi** [comment]
Actually, what I am thinking, you can go with the 2nd option, as I may not get much time to research on this, including, maintaining the same package on different platforms seems like a hectic continuous maintenance task as it seems they have their own new & unique configuration for each of them as I got to know while researching about the Homebrew package manager (#4). Also, it seems I need to have an Arch-based distro to use `makepkg` command if I'm not wrong.
So if you go with the 2nd option, i.e., putting the package on AUR and maintaining it yourself, before that, pls make sure to showcase what changes or config files you're putting on the AUR (like `PKGBUILD`, `.SRCINFO`, etc..) (at least for the initial push on the AUR repo). Thanks for the patience!
